qutebrowser/misc/guiprocess.py

```diff
@@ -109,13 +117,22 @@ def on_finished(self, code, status):
             message.error("{} exited with status {}, see :messages for "
                           "details.".format(self._what.capitalize(), code))
 
-            stderr = bytes(self._proc.readAllStandardError()).decode('utf-8')
-            stdout = bytes(self._proc.readAllStandardOutput()).decode('utf-8')
             if stdout:
                 log.procs.error("Process stdout:\n" + stdout.strip())
             if stderr:
                 log.procs.error("Process stderr:\n" + stderr.strip())
 
+    def spawn_format(self, code=0, status=0, stdout="", stderr=""):
```

Comment:
Ah, that's a clever solution to the issue with the tests! 👍 

However, I'd prefer to make this "private" (i.e. call it `_spawn_format`), as it's not used outside of tests.